packages/datasource-customizer/src/decorators/validation/action.ts

```diff
@@ -0,0 +1,58 @@
+import Ajv from 'ajv';
+import ajvErrors from 'ajv-errors';
+import ajvKeywords from 'ajv-keywords';
+
+import {
+  ActionConfigurationValidationError,
+  ActionFieldConfigurationValidationError,
+} from '../../errors';
+import { ActionDefinition, actionSchema } from '../actions/types/actions';
+import { DynamicField, fieldActionSchema } from '../actions/types/fields';
+
+const ajv = new Ajv({ allErrors: true });
+ajvErrors(ajv);
+ajvKeywords(ajv);
+
+export default class ActionValidator {
+  static validateActionConfiguration(name: string, action: ActionDefinition) {
+    const validate = ajv.compile(actionSchema);
+    if (!validate(action))
+      throw new ActionConfigurationValidationError(
+        name,
+        this.getValidationErrorMessage(validate.errors),
+      );
+
+    try {
+      action.form?.forEach(field => {
+        this.validateActionFieldConfiguration(field);
+      });
+    } catch (error) {
+      if (error instanceof ActionFieldConfigurationValidationError) {
+        throw new ActionConfigurationValidationError(name, `\n${error.message}`);
```

Comment:
```suggestion
        throw new ActionConfigurationValidationError(name, `${error.message}`);
```

Why adding the carriage return in there?

Reply:
It was an error. Should be fixed now